File: backend/app/schemas/quote.py
"""Quote schemas."""

from pydantic import BaseModel


class QuoteBase(BaseModel):
    """Base quote schema."""

    date: str
    open: float
    high: float
    low: float
    close: float
    volume: float | None = None
    turnover: float | None = None
    turnover_rate: float | None = None
    pe_ttm: float | None = None
    pb: float | None = None
    market_cap: float | None = None


class QuoteResponse(QuoteBase):
    """Quote response schema."""

    symbol: str

    class Config:
        from_attributes = True


class QuotesResponse(BaseModel):
    """Quotes list response."""

    quotes: list[QuoteBase]

File: frontend/src/types/quote.ts
export interface Quote {
  date: string
  open: number
  high: number
  low: number
  close: number
  volume: number | null
  turnover: number | null
  turnover_rate: number | null
  pe_ttm: number | null
  pb: number | null
  market_cap: number | null
}

export interface QuotesResponse {
  quotes: Quote[]
}
